climada/engine/impact.py

```diff
@@ -48,6 +47,410 @@
 
 LOGGER = logging.getLogger(__name__)
 
+class ImpactCalc():
+    """
+    Class to compute impacts from exposures, impact function set and hazard
+    """
+
+    def __init__(self,
+                 exposures,
+                 impfset,
+                 hazard,
+                 imp_mat=None):
+        """
+        Initialize an ImpactCalc object.
+
+        The dimension of the imp_mat variable must be compatible with the
+        exposures and hazard objects.
+
+        Parameters
+        ----------
+        exposures : climada.entity.Exposures
+            exposure used to compute imp_mat
+        impf_set: climada.entity.ImpactFuncSet
```

Comment:
`impfset` is actually written without underscore, I'm not sure what the intended notation is
```suggestion
        impfset: climada.entity.ImpactFuncSet
```

Reply:
I don't have strong feelings about what is best. Note that this is a change compared to the previous code where `impact_funcs` was used. I found this very confusing since we explicitly require a set and not a single function. Thus either `impf_set` or `impfset`. What do you prefer?

Reply:
I think I prefer `impf_set`